lib/futurism/channel.rb

```diff
@@ -15,17 +16,17 @@ def subscribed
     end
 
     def receive(data)
-      resources = data.fetch_values("signed_params", "sgids") { |key| [nil] }.transpose
-
+      resources = data.fetch_values("signed_params", "sgids", "signed_controllers") { |key|; Array.new(data["signed_params"].length, nil) }.transpose
```

Comment:
I'm not certain this will work in every case, but I believe it will

Reply:
I think it should. `data["signed_params"]` should always be set. Thinking about a way we can assert it, though. I'll add it to my bucket list!

Reply:
After seeing this comment, I looked around for "signed_params" and came across https://github.com/julianrubisch/futurism/blob/d2509b30189d53f848b7125794b144b6b52f0056/javascript/futurism_channel.js#L25-L29

I'm wondering if you'd be ok if we changed the code to something like:
```javascript
 debounceEvents(events => { 
   this.send(
    events.map(event => {
     const { dataset } = event.target
     return {
       signed_params: dataset.signedParams, 
       sgid: dataset.sgid,
       signed_controller: dataset.signedController
     })
  }) 
```

This would require quite a few changes to the channel and tests, but it's a clearer structure and doesn't have the interleaving require.

I'll say that I missed this entirely, so this whole PR doesn't work with passing controllers right now because I didn't catch that I needed to thread these data params through. Also, the communication between the front end and the channel back end was the biggest head-scratcher for me on this PR.

Does this new structure seem reasonable @julianrubisch?

Reply:
If it does, I might make a separate PR that changes the structure first and then come back to this one.

I could rip out the "default controller" feature as a separate PR to get that live to fix #54 in accordance with @scottbarrow's needs earlier as well.